src/System.Management.Automation/engine/CommandCompletion/ExtensibleCompletion.cs

```diff
@@ -200,38 +200,59 @@ public class RegisterArgumentCompleterCommand : PSCmdlet
         protected override void EndProcessing()
         {
             Dictionary<string, ScriptBlock> completerDictionary;
-            if (ParameterName != null)
+
+            if (ParameterName is null)
             {
-                completerDictionary = Context.CustomArgumentCompleters ??
-                                      (Context.CustomArgumentCompleters = new Dictionary<string, ScriptBlock>(StringComparer.OrdinalIgnoreCase));
+                completerDictionary = Context.NativeArgumentCompleters ??= new(StringComparer.OrdinalIgnoreCase);
+
+                foreach (string command in CommandName)
+                {
+                    var key = command?.Trim();
+                    if (string.IsNullOrEmpty(key))
+                    {
+                        continue;
+                    }
+
+                    SetKeyValue(completerDictionary, key, ScriptBlock);
```

Comment:
```suggestion
                    var key = command?.Trim();
                    if (!string.IsNullOrEmpty(key))
                    {
                        SetKeyValue(completerDictionary, key, ScriptBlock);
                    }
```

Reply:
"Exit early" is a common pattern to simplify maintenance. Not uncommon for folks to add code later and not realize it should be inside the if statement, with the current code you can always just add more lines with slightly less chance to mess it up

Reply:
Yes, if there are 200 lines below, but we only have one here.